### PR TITLE
[sdb] Add ability to retrieve custom attributes on assemblies

### PR DESCRIPTION
**Pull request by jbevain**
This reuses the existing pattern in Mono.Debugger.Soft to retrieve custom attributes.

The only questionable thing is moving the `buffer_add_cattrs` function instead of adding a forward declaration.

### CODE REVIEW

File: mono/mini/debugger-agent.c [hunk 1]
```diff
@@ -6853,6 +6854,114 @@ get_types_for_source_file (gpointer key, gpointer value, gpointer user_data)
 	}
 }
 
+
```
Comment (vargaz) [inline mono/mini/debugger-agent.c:6857]: Extra empty line.

**Review by vargaz — APPROVED** [review]

**Review by thaystg — APPROVED** [review]

File: mono/mini/debugger-agent.c (hunk 1 above)
Comment (jbevain) [reply mono/mini/debugger-agent.c]: Good catch, thanks Zoltan.

**Comment by vargaz** [comment]
@monojenkins commit apidiff

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by jbevain** [comment]
Test failure looks unrelated.

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**